### PR TITLE
ddl: exit add index on generated column with `case-when` expression parse error

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: AilinKid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/19325

### What is changed and how it works?
check the ddl job cancel state on portal of adding index every turn.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- ddl: exit add index on generated column with `case-when` expression parse error

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -1411,16 +1411,24 @@ func (w *worker) addPhysicalTableIndex(t table.PhysicalTable, indexInfo *model.I
 	logutil.BgLogger().Info("[ddl] start to add table index", zap.String("job", job.String()), zap.String("reorgInfo", reorgInfo.String()))
 	totalAddedCount := job.GetRowCount()
 
+	// It is used to close #19325 in earlier versions.
```
Comment (wjhuang2016) [inline ddl/index.go:1414]: I think we can remove this comment, or use a more understandable comment. Can we expect the reader to open github and search for this issue?

Comment (AilinKid) [reply ddl/index.go:1414]: ok

**Pull request merged**